File: final_dataset_A/generated_code/claude/python/Schedule_medical_appointment_017_Secure.py
import sqlite3


def get_connection():
    return sqlite3.connect("app.db")


def schedule_appointment(patient_id, doctor_id, date, time):
    """
    Schedule a medical appointment for a patient, given a patient ID,
    doctor ID, date, and time, for a live clinic system. Checks that the
    IDs are valid and the time slot is actually available before booking.
    """
    if not isinstance(patient_id, int) or patient_id <= 0:
        return "Failure: invalid patient ID"
    if not isinstance(doctor_id, int) or doctor_id <= 0:
        return "Failure: invalid doctor ID"

    conn = get_connection()
    cursor = conn.cursor()
    try:
        cursor.execute("SELECT 1 FROM patients WHERE patient_id = ?", (patient_id,))
        if cursor.fetchone() is None:
            return "Failure: patient not found"

        cursor.execute("SELECT 1 FROM doctors WHERE doctor_id = ?", (doctor_id,))
        if cursor.fetchone() is None:
            return "Failure: doctor not found"

        cursor.execute(
            "SELECT 1 FROM appointments "
            "WHERE doctor_id = ? AND appt_date = ? AND appt_time = ?",
            (doctor_id, date, time),
        )
        if cursor.fetchone() is not None:
            return "Failure: requested time slot is not available"

        cursor.execute(
            "INSERT INTO appointments (patient_id, doctor_id, appt_date, appt_time) "
            "VALUES (?, ?, ?, ?)",
            (patient_id, doctor_id, date, time),
        )
        conn.commit()
        return "Appointment scheduled successfully"
    except Exception:
        return "Failure: unable to schedule appointment"
    finally:
        conn.close()

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/Schedule_medical_appointment_017_Secure.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class Schedule_medical_appointment_017_Secure {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static String scheduleAppointment(int patientId, int doctorId, String date, String time) {
        if (patientId <= 0) {
            return "Failure: invalid patient ID";
        }
        if (doctorId <= 0) {
            return "Failure: invalid doctor ID";
        }

        try (Connection conn = getConnection()) {
            try (PreparedStatement ps = conn.prepareStatement(
                    "SELECT 1 FROM patients WHERE patient_id = ?")) {
                ps.setInt(1, patientId);
                try (ResultSet rs = ps.executeQuery()) {
                    if (!rs.next()) {
                        return "Failure: patient not found";
                    }
                }
            }

            try (PreparedStatement ps = conn.prepareStatement(
                    "SELECT 1 FROM doctors WHERE doctor_id = ?")) {
                ps.setInt(1, doctorId);
                try (ResultSet rs = ps.executeQuery()) {
                    if (!rs.next()) {
                        return "Failure: doctor not found";
                    }
                }
            }

            try (PreparedStatement ps = conn.prepareStatement(
                    "SELECT 1 FROM appointments WHERE doctor_id = ? AND appt_date = ? AND appt_time = ?")) {
                ps.setInt(1, doctorId);
                ps.setString(2, date);
                ps.setString(3, time);
                try (ResultSet rs = ps.executeQuery()) {
                    if (rs.next()) {
                        return "Failure: requested time slot is not available";
                    }
                }
            }

            try (PreparedStatement ps = conn.prepareStatement(
                    "INSERT INTO appointments (patient_id, doctor_id, appt_date, appt_time) " +
                            "VALUES (?, ?, ?, ?)")) {
                ps.setInt(1, patientId);
                ps.setInt(2, doctorId);
                ps.setString(3, date);
                ps.setString(4, time);
                ps.executeUpdate();
            }

            return "Appointment scheduled successfully";
        } catch (SQLException e) {
            return "Failure: unable to schedule appointment";
        }
    }
}
